source/AI.cpp

```diff
@@ -3824,21 +3923,22 @@ void AI::IssueOrders(const PlayerInfo &player, const Orders &newOrders, const st
 
 	// A target is valid if we have no target, or when the target is in the
 	// same system as the flagship.
-	bool isValidTarget = !newTarget || (newTarget && player.Flagship() &&
-		newTarget->GetSystem() == player.Flagship()->GetSystem());
+	bool isValidTarget = !targetShip || targetAsteroid
```

Comment:
The comment suggests a mismatch between the original logic and the current logic
"A target is valid if we have no target..."

Reply:
The comment should probably be updated if it's not clear.

I felt the original comment was accurate

> A target is valid... when the target is in the same system as the flagship.

Because if an asteroid is not null then it is in the same system as the flagship 100% of the time.  When the player jumps out of the system `Clear()` function is called which removes the target.

Reply:
I think the comment is clear; let me know if you feel strongly for change.  I'll resolve convo for now.